jsettlers.logic/src/jsettlers/logic/map/original/OriginalMapFileContentReader.java

```diff
@@ -0,0 +1,759 @@
+		int pos = FPart.offset;
+		
+		//- Number of buildings
+		int SettlerCount = readBEIntFrom(pos);
+		pos += 4;
+		
+		//- safety check
+		if ((SettlerCount * 6 > FPart.size) || (SettlerCount < 0)) {
+			System.err.println("wrong number of settlers in map File: "+ SettlerCount);
+			SettlerCount = 0;
+		    return false;
+		}
+		
+		
+		//- read all Stacks
+		for (int i = 0 ; i < SettlerCount; i++) {
+		 
+			int party = readByteFrom(pos++);
+			int SType = readByteFrom(pos++);
+			
+			int x_pos = readBEWordFrom(pos);  pos+=2;
+			int y_pos = readBEWordFrom(pos);  pos+=2;
+		
+			
+			//-------------
+	        //- update data                              
+			mapData.setSettler(x_pos, y_pos, SType, party);
+		}
+		
+		return true;
+	}
+	
+	
+	//- Read the Player Info
+	public void readPlayerInfo() {
+		MapResourceInfo FPart = findResource(OriginalMapFileDataStructs.EMapFilePartType.PLAYER_INFO);
+		
+		if ((FPart == null) || (FPart.size == 0)) {
+			System.err.println("Warning: No Player information available in mapfile!");
+			return;
+		}
+
+		//- Decrypt this resource if necessary
+		if (!doDecrypt(FPart)) return;
+
+		//- file position
+		int pos = FPart.offset;
+		
+		for (int i = 0; i < mapData.getPlayerCount(); i++) {
+			
+			int nation = readBEIntFrom(pos);
+			pos += 4;
+			
+			int startX = readBEIntFrom(pos);
+			pos += 4;
+			
+			int startY = readBEIntFrom(pos);
+			pos += 4;
+			
+			String playerName = readCStrFrom(pos, 33);
+			pos += 33;
+			
+			mapData.setPlayer(i, startX, startY, nation, playerName);
+		
+		}
+	}
+	
+	
+	//- Reads in the Map Data / Landscape and MapObjects like trees
+	public boolean readMapData() {
+		
+		//- get resource information for the area 
+		MapResourceInfo FPart = findResource(OriginalMapFileDataStructs.EMapFilePartType.AREA);
+		
+		if ((FPart==null) || (FPart.size == 0)) {
+			System.err.println("Warning: No area information available in mapfile!");
+			return false;
+		}
+
+		//- Decrypt this resource if necessary
+		if (!doDecrypt(FPart)) return false;
+
+		//- file position
+		int pos = FPart.offset;
+		
+		//- height and width are the same
+		int WidthHeight = readBEIntFrom(pos);
+		pos+=4;
+		
+		//- init size of MapData
+		mapData.setWidthHeight(WidthHeight);
+
+		//- points to read
+		int dataCount = WidthHeight * WidthHeight;
+		
+		for (int i=0; i< dataCount; i++) {
+			mapData.setLandscapeHeight(i, readByteFrom(pos++));
+			mapData.setLandscape(i, readByteFrom(pos++));
+			mapData.setMapObject(i, readByteFrom(pos++));
+			int PalyerClaim = readByteFrom(pos++); //- which Player is the owner of this position
+			mapData.setAccessible(i, mapContent[pos++]);
+			
+			mapData.setResources(i, readHighNibbleFrom(pos), readLowNibbleFrom(pos));
+			pos++;
+		}
+		
+
+		return true;
+	}
+
+
+	public void addStartTowerMaterialsAndSettlers() {
+		//- only if there are no buildings
+		if (hasBuildings) return;
+		
+		//- only do this once
+		if (startTowerMaterialsAndSettlersWereSet) return;	
```

Comment:
@tomsoftware for your information: I remove this check because with this check you can't load a game from the same original map twice.
It took me a while to find this. The AI tests crashed .. but the first always ran through and all following crashed due to multiple starting a game with the same original map. It turned out, that the start towers were missing.